src/Smalot/PdfParser/PDFObject.php

```diff
@@ -246,6 +246,36 @@ private function getDefaultFont(Page $page = null): Font
         return new Font($this->document, null, null, $this->config);
     }
 
+    private function getTJUsingFontFallback(Font $font, array $command, Page $page = null): string
```

Comment:
Third parameter seems invalid. If `$page` can be `null`, than it has to be `?Page $page = null`.

Also, how does array `$command` look like? Please have a look at https://phpstan.org/writing-php-code/phpdoc-types#general-arrays

If you are not sure, we can use the following for now:

 ```suggestion
    /*
     * @param array<mixed> $command
     */
    private function getTJUsingFontFallback(Font $font, array $command, ?Page $page = null): string
```

Reply:
$command looks like this. It's an array of parsed elements from a single line of stream content.

```
array(31) {
  [0]=>
  array(3) {
    ["t"]=>
    string(1) "<"
    ["o"]=>
    string(0) ""
    ["c"]=>
    string(4) "023E"
  }
  [1]=>
  array(3) {
    ["t"]=>
    string(1) "n"
    ["o"]=>
    string(0) ""
    ["c"]=>
    string(3) "1.7"
  }
  [2]=>
  array(3) {
    ["t"]=>
    string(1) "<"
    ["o"]=>
    string(0) ""
    ["c"]=>
    string(4) "025F"
  }
  [3]=>
  array(3) {
    ["t"]=>
    string(1) "n"
    ["o"]=>
    string(0) ""
    ["c"]=>
    string(3) "2.9"
  }
  ...
```

So how would you write that? Maybe `list<array>`?

Reply:
Based on the given data it is `array<int,array<string,string>>` in my opinion. If values can be `null`, it must be `array<int,array<string,string|null>>` instead.

Reply:
I think it might be `array<int,array<string,mixed>>`. If you look at `getCommandsText()` it initializes the three values that end up in "t", "o" and "c" as `string`, `string`, `bool` respectively.

```php
            $operator = '';
            $type = '';
            $command = false;
```

If I change it to `$command = '';` the output is broken so the `bool` is required here somehow.

Reply:
I am not familiar with the content of those variables. Its just a minor thing, but `array<int,array<string,string|bool>>` would be most precise. But either way, use one that fits the best in your opinion.